tests/test_polynomial.py

```
@@ -0,0 +1,988 @@
+        # Should mask duplicates (after sort(), masked values become inf)
+        self.assertEqual(roots_dup_array.shape, (2, 2, 2))
+        self.assertTrue(np.any(~np.isfinite(roots_dup_array.values)))
+
+        # Additional tests for 100% coverage
+
+        # Test __iadd__ when arg needs set_order (line 263)
+        p_iadd1 = Polynomial([1., 2.])  # order 1
+        p_iadd2 = Polynomial([3., 4., 5.])  # order 2
+        id_before = id(p_iadd1)
+        p_iadd1 += p_iadd2
+        self.assertEqual(id(p_iadd1), id_before)  # In-place
```

Ruff (0.14.7)

Use a regular `assert` instead of unittest-style `assertEqual`

Replace `assertEqual(...)` with `assert ...`

(PT009)